pkgs/development/libraries/dlib/default.nix

```diff
@@ -2,7 +2,7 @@
 , guiSupport ? false, libX11
 
   # see http://dlib.net/compile.html
-, avxSupport ? true
+, avxSupport ? builtins.elem (stdenv.hostPlatform.platform.gcc.arch or "default") [ "sandybridge" "ivybridge" "haswell" "broadwell" "skylake" "skylake-avx512" "btver2" "bdver1" "bdver2" "bdver3" "bdver4" "znver1"]
```

Comment:
I feel like this could be nicer to work with if it was something like `stdenv.hostPlatform.platform.gcc.arch.hasAvxSupport`, where gcc knows whether avx is supported

Reply:
`stdenv.hostPlatform.platform` is a plain attrset, without any functions, so I added the functions into `stdenv.hostPlatform`, next to `.isx86_64` and other predicates